libbeat/processors/add_kubernetes_metadata/config.go

```diff
@@ -69,5 +69,34 @@ func (k *kubeAnnotatorConfig) Validate() error {
 		k.Host = ""
 	}
 
+	for _, matcher := range k.Matchers {
```

Comment:
Looks good! I would suggest adding a comment here explaining why we perform such checks and that it's because we are taking an opinionated approach on what paths we support.

Reply:
https://github.com/elastic/beats/pull/28868/commits/2dfe013789a4ee5f396881ae83808cd6b76a90db